logger/logger.c

```diff
@@ -217,12 +216,27 @@ void logger_stop ()
 	}
 }
 
-void log_runtime (log_level_t level, char *msg)
-{
+/* 
+ * Logs a message at a specified level to the locations specified in config file
+ * Handles format strings (can handle expressions like those in 'printf()')
+ * Arguments:
+ *    - log_level_t level: one of the levels listed in the enum in this file
+ *    - char *format: format string representing message to be formatted
+ *    - ...: additional arguments to be formatted in format string
+ */
+void log_printf (log_level_t level, char *format, ...)
+{	
 	static time_t now;                   //for holding system time
 	static char *time_str;               //for string representation of system time
 	static char final_msg[MAX_LOG_LEN];  //final message to be logged to requested locations
 	static int len;                      //holding lengths of various strings
+	static char msg[MAX_LOG_LEN - 100];  //holds the expanded format string (100 to make room for log header)
+	va_list args;                        //this holds the variable-length argument list
+	
+	//expands the format string into msg
+	va_start(args, format);
+	vsprintf(msg, format, args);
```

Comment:
probably should put this after you check if the level is high enough

Reply:
Good point. Fixed